### PR TITLE
[Cult 3.0] mostly Cult Altar related stuff

### PR DESCRIPTION
**Pull request by DeityLink**
* Fixed a few bugs (such as Communication rune working with anyone, transmitting to everyone; or mobs not getting their antag icons if given a faction at the same time as a role).
* Streamlined cult structures so they can share procs regarding interactions, and damage.
* Raise Structure rune is now working, though it will only spawn cult altars. Progress bar integrated, other cultists can click the rune/vortex to take part in the ritual.
* Added Cult Altar, doesn't do much for now, but can be interacted with. Also you can use it like a table, and drag people on top of it.
* get_unarmed_damage() procs now take atoms instead of mobs as arg so they can be used against non-mobs.

image

I'm open to suggestions for the Raise Structure rune's incantation. Preferably something believable.

### CODE REVIEW

**Comment by DrSnips** [comment]
Possible chant
CON'ST A DI'NAL PHI LONS
Maybe change the end bit depending on the structure summoned.

**Comment by DeityLink** [comment]
Not enough consonant. Here's an example of what I'm looking for:
* "Ta'gh fara'qha fel d'amar det!"
* "Khari'd! Gual'te nikka!"
* "Nikt'o barada kla'atu!"
* "Sti' kaliedir!"

The languages in lovecraftian lore involve a lot of tongue clackings. It's fine if it's just gibberish without meaning, the sonority is more important.

**Comment by Blithering** [comment]
Kaa'n Strahk'tin?

**Comment by DeityLink** [comment]
I like the pun. I might have something similar, along with some extras depending on the building like Snips suggested.

I think I'll add them during the polish phase though.

File: code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+		cultist_act(user)
+	else
+		noncultist_act(user)
+
+/obj/structure/cult/proc/cultist_act(var/mob/user)
+	if(!iscultist(user))//just to be extra safe
+		return 0
+	return 1
+
+/obj/structure/cult/proc/noncultist_act(var/mob/user)
```
Comment (MadmanMartian) [inline code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm:94]: Remember the cultist chaplain religion, could be fun for here

Comment (DeityLink) [reply code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm:94]: Yeah I made a reference to it in a comment like 4 lines bellow. There'll be some polish once every Cult 3.0 feature is functional.

File: code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+	playsound(get_turf(src), sound_damaged, 75, 1)
+	takeDamage(20)
+
+/obj/structure/cult/bullet_act(var/obj/item/projectile/Proj)
+	takeDamage(Proj.damage)
+	..()
+
+/obj/structure/cult/attackby(var/obj/item/weapon/W, var/mob/user)
+	if (istype(W, /obj/item/weapon/grab))
+		if(iscarbon(W:affecting))
```
Comment (MadmanMartian) [inline code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm]: Why the colons?

File: code/modules/mob/living/carbon/alien/humanoid/combat.dm [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 	return "slashes at"
 
-/mob/living/carbon/alien/humanoid/get_unarmed_damage(mob/living/target)
+/mob/living/carbon/alien/humanoid/get_unarmed_damage(var/atom/target)
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/alien/humanoid/combat.dm:38]: why these changes to this and the other get_unarmed_damages?

File: code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm (hunk 2 above)
Comment (DeityLink) [reply code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm]: fixed

File: code/modules/mob/living/carbon/alien/humanoid/combat.dm (hunk 3 above)
Comment (Probe1) [reply code/modules/mob/living/carbon/alien/humanoid/combat.dm:38]: @DeityLink pls respond

Comment (MadmanMartian) [reply code/modules/mob/living/carbon/alien/humanoid/combat.dm:38]: >>get_unarmed_damage() procs now take atoms instead of mobs as arg so they can be used against non-mobs.

I missed it, he informed me of this, and I forgot to remove this, sorry.
